metals/src/main/scala/scala/meta/internal/metals/BuildTargetMapper.scala

```diff
@@ -0,0 +1,41 @@
+package scala.meta.internal.metals
+
+import scala.collection.concurrent.TrieMap
+import scala.collection.mutable
+
+import ch.epfl.scala.bsp4j.BuildTargetIdentifier
+
+class BuildTargetMapper {
```

Comment:
Not perfect, but I don't have a better name. I also added a comment, because the whole logic may not be that obvious. Can you check if it is understandable?